Mono.Unix is not available by default in macOS

### Checklist

- [x] I have checked the [issue page](https://github.com/lay295/TwitchDownloader/issues?q=is%3Aissue) for duplicates
- [x] I am running the latest version ([download here](https://github.com/lay295/TwitchDownloader/releases/latest))

### Edition

Command Line Interface

### Describe your issue here

I'm trying to update the CLI but Mono can't be expected to be available in all macOS systems:

```
$ TwitchDownloaderCLI update
TwitchDownloaderCLI 1.55.5 Copyright (c) lay295 and contributors
[INFO] - A new version of TwitchDownloaderCLI is available (v1.55.6)!
Would you like to update?
[Y] Yes / [N] No:
Y
[STATUS] - Downloading Update 100%Unhandled exception. System.DllNotFoundException: Unable to load shared library 'MonoPosixHelper' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(libMonoPosixHelper, 0x0001): tried: 'libMonoPosixHelper' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibMonoPosixHelper' (no such file), '/usr/lib/libMonoPosixHelper' (no such file, not in dyld cache), 'libMonoPosixHelper' (no such file), '/usr/local/lib/libMonoPosixHelper' (no such file), '/usr/lib/libMonoPosixHelper' (no such file, not in dyld cache)
   at Mono.Unix.Native.Syscall.stat(String file_name, Stat& buf)
   at Mono.Unix.UnixFileSystemInfo.GetFileStatus(String path, Stat& stat)
   at Mono.Unix.UnixFileSystemInfo.Refresh(Boolean force)
   at Mono.Unix.UnixFileSystemInfo..ctor(String path)
   at Mono.Unix.UnixFileInfo..ctor(String path)
   at TwitchDownloaderCLI.Modes.UpdateHandler.AutoUpdate(String url, ITaskProgress progress)
   at TwitchDownloaderCLI.Modes.UpdateHandler.AutoUpdate(String url, ITaskProgress progress)
   at TwitchDownloaderCLI.Modes.UpdateHandler.CheckForUpdate(UpdateArgs args, ITaskProgress progress)
   at TwitchDownloaderCLI.Modes.UpdateHandler.ParseArgs(UpdateArgs args)
   at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
   at TwitchDownloaderCLI.Program.Main(String[] args)
```

Also notice that the error is launched in the same line as the progress report `100%Unhandled exception.`.

### Add any related files or extra information here

_No response_

I use the Ffmpeg in my system, but fails when trying to download it:

```
$ TwitchDownloaderCLI ffmpeg -d --log-level Info,Verbose,Status,Warning,Error,Ffmpeg
TwitchDownloaderCLI 1.55.6 Copyright (c) lay295 and contributors
[STATUS] - Downloading FFmpeg 100%

[ERROR] - Unable to automatically update FFmpeg file permissions. Please run "chmod +x ffmpeg" to allow FFmpeg to be executed.
```

```
$ TwitchDownloaderCLI ffmpeg -d --log-level Info,Verbose,Status,Warning,Error,Ffmpeg
TwitchDownloaderCLI 1.55.6 Copyright (c) lay295 and contributors
[INFO] - FFmpeg was already found in the current working directory.
```

Notice that there is an empty line there. Also I don't like that doesn't tell where the ffmpeg is saved to.

--- a/TwitchDownloaderCLI/Modes/FfmpegHandler.cs
+++ b/TwitchDownloaderCLI/Modes/FfmpegHandler.cs
@@ -50,8 +50,7 @@ namespace TwitchDownloaderCLI.Modes
             {
                 var ffmpegFileInfo = new UnixFileInfo("ffmpeg")
                 {
-                    FileAccessPermissions = FileAccessPermissions.UserRead | FileAccessPermissions.UserWrite | FileAccessPermissions.GroupRead | FileAccessPermissions.OtherRead |
-                                            FileAccessPermissions.UserExecute | FileAccessPermissions.GroupExecute | FileAccessPermissions.OtherExecute
+                    FileAccessPermissions = FileAccessPermissions.UserReadWriteExecute | FileAccessPermissions.GroupReadWriteExecute | FileAccessPermissions.OtherReadWriteExecute,
                 };
                 ffmpegFileInfo.Refresh();
             }

--- a/TwitchDownloaderCLI/Modes/UpdateHandler.cs
+++ b/TwitchDownloaderCLI/Modes/UpdateHandler.cs
@@ -179,9 +179,8 @@ namespace TwitchDownloaderCLI.Modes
             }
 
             // Get old unix file permissions
-            var isUnx = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
-            var previousPermissions = FileAccessPermissions.DefaultPermissions;
-            if (isUnx)
+            FileAccessPermissions? previousPermissions = null;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
             {
                 try
                 {
@@ -221,13 +220,13 @@ namespace TwitchDownloaderCLI.Modes
             File.Delete(archivePath);
 
             // Apply previous file permissions
-            if (isUnx)
+            if (previousPermissions.HasValue)
             {
                 try
                 {
                     var ufi = new UnixFileInfo(currentExePath)
                     {
-                        FileAccessPermissions = previousPermissions
+                        FileAccessPermissions = previousPermissions.Value
                     };
                     ufi.Refresh();
                 }

--- a/TwitchDownloaderCLI/TwitchDownloaderCLI.csproj
+++ b/TwitchDownloaderCLI/TwitchDownloaderCLI.csproj
@@ -13,7 +13,7 @@
 
   <ItemGroup>
     <PackageReference Include="CommandLineParser" Version="2.9.1" />
-    <PackageReference Include="Mono.Posix.NETStandard" Version="1.0.0" />
+    <PackageReference Include="Mono.Posix" Version="7.1.0-final.1.21458.1" />
     <PackageReference Include="Spectre.Console" Version="0.49.1" />
     <PackageReference Include="Xabe.FFmpeg.Downloader" Version="5.2.6" />
   </ItemGroup>

--- a/TwitchDownloaderCore/TwitchDownloaderCore.csproj
+++ b/TwitchDownloaderCore/TwitchDownloaderCore.csproj
@@ -25,7 +25,7 @@
   <ItemGroup>
     <PackageReference Include="HarfBuzzSharp" Version="7.3.0.3" />
     <PackageReference Include="HarfBuzzSharp.NativeAssets.Linux" Version="7.3.0.3" />
-    <PackageReference Include="Mono.Posix.NETStandard" Version="1.0.0" />
+    <PackageReference Include="Mono.Posix" Version="7.1.0-final.1.21458.1" />
     <PackageReference Include="SkiaSharp" Version="2.88.9" />
     <PackageReference Include="SkiaSharp.HarfBuzz" Version="2.88.9" />
     <PackageReference Include="SkiaSharp.NativeAssets.Linux" Version="2.88.9" />

--- a/TwitchDownloaderCore/TwitchHelper.cs
+++ b/TwitchDownloaderCore/TwitchHelper.cs
@@ -9,11 +9,13 @@ using System.Net;
 using System.Net.Http;
 using System.Net.Http.Json;
 using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
 using System.Security;
 using System.Text;
 using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
+using Mono.Unix;
 using TwitchDownloaderCore.Chat;
 using TwitchDownloaderCore.Interfaces;
 using TwitchDownloaderCore.Tools;
@@ -993,10 +995,14 @@ namespace TwitchDownloaderCore
             return directoryInfo;
         }
 
-        public static void SetDirectoryPermissions(string path)
+        [SupportedOSPlatform("linux")]
+        [SupportedOSPlatform("osx")]
+        private static void SetDirectoryPermissions(string path)
         {
-            var folderInfo = new Mono.Unix.UnixFileInfo(path);
-            folderInfo.FileAccessPermissions = Mono.Unix.FileAccessPermissions.AllPermissions;
+            var folderInfo = new UnixFileInfo(path);
+            folderInfo.FileAccessPermissions = FileAccessPermissions.UserReadWriteExecute |
+                                               FileAccessPermissions.GroupRead | FileAccessPermissions.GroupWrite |
+                                               FileAccessPermissions.OtherRead | FileAccessPermissions.OtherWrite;
             folderInfo.Refresh();
         }
 

